spec/system/admin_rewards_from_csv_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'rails_helper'
+require 'factory_bot_rails'
+
+RSpec.describe 'Import rewards from a CSV', type: :system do
+  let(:admin_user) { create(:admin_user) }
+
+  before do
+    driven_by(:rack_test)
+    sign_in admin_user
+  end
+
+  context 'when importing rewards from csv file' do
+    it 'imports reward from csv file' do
+      expect(Reward.count).to eq 0
+      visit dashboard_admin_users_pages_path
+      click_link 'Manage Rewards'
+
+      click_link 'Import rewards from CSV'
+      attach_file(File.absolute_path('./spec/files/test1.csv'))
```

Comment:
For such imports, try to import more than one row/record to be sure loop works correctly.

Reply:
Done!